OpenRA.Mods.Common/Traits/Crates/Crate.cs

```diff
@@ -144,7 +144,7 @@ public void SetPosition(Actor self, WPos pos)
 		{
 			var cell = self.World.Map.CellContaining(pos);
 			SetLocation(self, cell);
-			SetVisualPosition(self, self.World.Map.CenterOfCell(cell) + new WVec(0, 0, pos.Z));
+			SetVisualPosition(self, self.World.Map.CenterOfCell(cell) + new WVec(WDist.Zero, WDist.Zero, self.World.Map.DistanceAboveTerrain(pos)));
```

Comment:
The CPos version has no way to offset from the cell's CenterPosition - CenterOfCell returns the cell's Z offset afterall, the bug doesn't exist there.